	2009-09-02  Daniel Jacobowitz  <dan@codesourcery.com>

	libgcc/
	* shared-object.mk (c_flags-$(base)$(objext)): New.
	($(base)$(objext)): Use above.
	($(base)_s$(objext)): Likewise.
	* static-object.mk (c_flags-$(base)$(objext)): New.
	($(base)$(objext)): Use above.
	
2010-07-10  Yao Qi  <yao@codesourcery.com>

	Merge from Sourcery G++ 4.4:

 	2009-05-28  Julian Brown  <julian@codesourcery.com>
 
 	Merged from Sourcery G++ 4.3:

=== modified file 'libgcc/shared-object.mk'
--- old/libgcc/shared-object.mk	2008-07-03 18:22:00 +0000
+++ new/libgcc/shared-object.mk	2010-07-30 13:11:02 +0000
@@ -8,11 +8,13 @@
 
 ifeq ($(suffix $o),.c)
 
+c_flags-$(base)$(objext) := $(c_flags)
 $(base)$(objext): $o
-	$(gcc_compile) $(c_flags) -c $< $(vis_hide)
+	$(gcc_compile) $(c_flags-$@) -c $< $(vis_hide)
 
+c_flags-$(base)_s$(objext) := $(c_flags)
 $(base)_s$(objext): $o
-	$(gcc_s_compile) $(c_flags) -c $<
+	$(gcc_s_compile) $(c_flags-$@) -c $<
 
 else
 

=== modified file 'libgcc/static-object.mk'
--- old/libgcc/static-object.mk	2007-01-04 04:22:37 +0000
+++ new/libgcc/static-object.mk	2010-07-30 13:11:02 +0000
@@ -8,8 +8,9 @@
 
 ifeq ($(suffix $o),.c)
 
+c_flags-$(base)$(objext) := $(c_flags)
 $(base)$(objext): $o
-	$(gcc_compile) $(c_flags) -c $< $(vis_hide)
+	$(gcc_compile) $(c_flags-$@) -c $< $(vis_hide)
 
 else
 

